docs/SodiumPlus/encrypted-streams.md

```diff
@@ -10,84 +10,65 @@ Initialize a stream for streaming encryption.
 
 1. `{CryptographyKey}` key
 
-Returns a `Promise` that resolves to an `array` with 2 elements:
+Returns a `Promise` that resolves to an object with following properties & methods:
 
-1. A 24-byte header that should be included in the encrypted stream.
-2. A backend-specific `state`:
-    * `LibsodiumWrappers` returns a `number` (a pointer to an internal buffer)
-    * `SodiumNative` returns a `CryptoSecretstreamXchacha20poly1305StateWrap`
-      object
+* `header`: a 24-byte header that should be included in the encrypted stream.
+* `push`: encrypt some data in a stream.
 
-The `{state}` type annotation below refers to one of the backend-specific state 
-types.
+    **Parameters and their respective types**:
 
-You'll typically want to use it with list unpacking syntax, like so:
+    1. `{string|Buffer}` message
+    1. `{string|Buffer}` (optional) additional associated data
+    1. `{number}` tag (default = 0, see libsodium docs)
 
-```
-[state, header] = await sodium.crypto_secretstream_xchacha20poly1305_init_push(key);
-```
+    Returns a `Promise` that resolves to a `Buffer` containing the ciphertext.
 
-### crypto_secretstream_xchacha20poly1305_init_pull()
+* `rekey`: deterministic re-keying of the internal state.
 
-Initialize a stream for streaming decryption.
+    Returns a `Promise` that resolves to `undefined`. Instead,
+    the underlying backend-specific `state` is overwritten in-place.
 
-**Parameters and their respective types**:
-
-1. `{CryptographyKey}` key
-2. `{Buffer}` header (must be 24 bytes)
-
-Returns a `Promise` that resolves to a backend-specific `state`:
+_Note:_ the object that the `Promise` is resolved to wraps following backend-specific state
+restricting direct access to it:
 
 * `LibsodiumWrappers` returns a `number` (a pointer to an internal buffer)
 * `SodiumNative` returns a `CryptoSecretstreamXchacha20poly1305StateWrap`
   object
 
-The `{state}` type annotation below refers to one of the backend-specific state 
-types.
-
-### crypto_secretstream_xchacha20poly1305_push()
```

Comment:
I am not yet 100% sure that moving detailed documentation for the `push` method from a separate section deep into the description of the value returned by the `crypto_secretstream_xchacha20poly1305_init_push` method is the best option.

My main concern is that now `crypto_secretstream_xchacha20poly1305_init_push` method description barely fits on a screen and, if some details are added in future, it will stop fitting completely. It also seems like description for the `push` and `pull` methods is not visually separated from documentation of the `init_push` and `init_pull` methods.
So as an alternative I am considering adding new _Stream Encryptor object_ and _Stream Decryptor object_ sub-sections describing properties and methods of the objects _asynchronously returned_ by the `crypto_secretstream_xchacha20poly1305_init_push` and `crypto_secretstream_xchacha20poly1305_init_pull` methods.

@paragonie-scott, @paragonie-security, what do you think of how this looks like now?
Please let me know if I should revise this using the alternative option.

This is how `crypto_secretstream_xchacha20poly1305_init_push` method description looks like with these changes:
![image](https://user-images.githubusercontent.com/2260810/88464893-066f6980-cec7-11ea-9092-00f7d020e24a.png)

Reply:
Here's how it looks like in the alternative option:
![image](https://user-images.githubusercontent.com/2260810/88465866-4c303000-cecf-11ea-95c5-69832aaea3b7.png)
![image](https://user-images.githubusercontent.com/2260810/88465870-59e5b580-cecf-11ea-945b-c1498d3509d0.png)